extensions/Khronos/KHR_materials_unlit/README.md

```diff
@@ -0,0 +1,118 @@
+
+## Overview
+
+This extension defines an unlit shading model for use in glTF 2.0
+materials, as an alternative to the Physically Based Rendering (PBR) shading
+models provided by the core specification. In mobile devices with limited
+resources, unlit materials offer a performant alternative to higher-quality
+shading models.
+
+## Extending Materials
+
+The common Unlit material is defined by adding the
+`KHR_materials_unlit` extension to any glTF material. When present, the
+extension indicates that a material should be unlit and use available
+`baseColor` values and vertex colors, ignoring other properties of the default
+PBR model.
+
+```json
+{
+    "materials": [
+        {
+            "name": "MyUnlitMaterial",
+            "pbrMetallicRoughness": {
+                "baseColorFactor": [ 0.5, 0.8, 0.0, 1.0 ]
+            },
+            "extensions": {
+                "KHR_materials_unlit": {}
+            }
+        }
+    ]
+}
+```
+
+### Definition
+
+The Unlit material model describes a constantly shaded surface that is
+independent of lighting. The material is defined only by properties already
+present in the [glTF 2.0 material specification](https://github.com/KhronosGroup/glTF/tree/master/specification/2.0#material).
+No new properties are added by this extension — it is effectively a boolean
+flag indicating use of an unlit shading model. Additional properties on the
+extension object are allowed, but may lead to undefined behaviour in conforming
+viewers.
+
+Color is calculated as:
+
+```
+color = <baseColorTerm>
+```
+
+`<baseColorTerm>` is defined by the [core glTF material specification](https://github.com/KhronosGroup/glTF/blob/master/specification/2.0/README.md#metallic-roughness-material).
+
+### Example
+
+The following example defines an unlit material with a constant red color.
+
+```json
+"materials": [
+    {
+        "name": "red_unlit_material",
+        "pbrMetallicRoughness": {
+            "baseColorFactor": [ 1.0, 0.0, 0.0, 1.0 ]
+        },
+        "extensions": {
+            "KHR_materials_unlit": {}
+        }
+    }
+]
+```
+
+### Extension compatibility and fallback materials
+
+When possible, authoring tools should provide a fallback material definition
+and mark the `KHR_materials_unlit` extension as optional. Models
+including the extension optionally will still render in all clients that
+support the core glTF 2.0 specification, falling back to a PBR metal/rough
+material. In clients that do not support the extension, direct lighting will
+be applied and performance may be degraded.
+
+Defining a fallback to the default PBR shading model is achieved by setting
+appropriate parameters on the default material.
+
+```json
+"materials": [
+    {
+        "name": "unlit_with_fallback",
+        "pbrMetallicRoughness": {
+            "baseColorFactor": [ 1.0, 1.0, 1.0, 1.0 ],
+            "baseColorTexture": { "index": 0 },
+            "metallicFactor": 0.0,
+            "roughnessFactor": 1.0
```

Comment:
`roughnessFactor` defaults to `1.0` and thus is not needed. I believe it will be flagged as a warning in the validator. Same goes for `baseColorFactor`, but I would recommend changing the values so that they are not the default like in the "red_unlit_material" above.

Reply:
I agree it would be better to choose a fallback example here that doesn't use extreme values, like roughness 1.0. The fallback materials shouldn't necessarily try to look as close to individually "unlit" as possible, right? Rather, we know the fallback materials are only used in the situation where we're in a fully lit, PBR environment. Better to accept and embrace that fact and pick a fallback material that looks attractively natural in such a setting.

In that vein, while keeping it simple, perhaps drop roughness to 0.8, include a normal map?

Reply:
I believe all the scenarios for an unlit material can be categorized into two buckets.

1. The unlit material is the primary material and the core PBR material is the fallback. In this case, if the client implementation can render unlit, it should render with unlit. The fallback core PBR material should be author to match as closely as possible to the simulate what an unlit material would do.

2. The core PBR material is the primary material and the unlit material is the fallback. In this case, if the client implementation can render core PBR metallic/roughness model, then it should render with PBR metallic/roughness model, even if there is an unlit extension. The fallback, likely due to runtime resources, is to render using unlit material as it is much cheaper to render. In this scenario, the core PBR material may have additional maps, various metallic/roughness values, etc. to support the higher fidelity material.

All of the scenarios that have been mentioned so far, including photogrammetry, seems to point to 1, but I think 2 is still possible and we shouldn't restrict someone from doing it.

Maybe we can incorporate this kind of language into this section? It would all be non-normative.

Reply:
> fallback core PBR material authored to match as closely as possible to simulate what an unlit material would do

To be clear, what I mean here is that the core PBR material should match what this PR currently already states:

> **Implementation Note:** For best fallback behavior in clients that do not
> implement the `KHR_materials_unlit` extension, authoring tools may use:
>
> * `metallicFactor` and `emissiveFactor` are `0`.
> * `roughnessFactor` is `1`.
> * Omit `metallicRoughnessTexture`, `occlusionTexture`, `emissiveTexture`,
> and `normalTexture`.

Reply:
This sounds good to me

Reply:
To add to this a bit more, in scenario 2, there is no reason to use the unlit extension at all. The client implementation can fallback to unlit if necessary on its own.

Reply:
I am content to withdraw my hesitation above re: roughness 1.0 vs 0.8 and so forth. It's fine for this spec to recommend a somewhat extreme fallback material.

Reply:
I removed `roughnessFactor=1.0` from the example, as that's the default value anyway.

Perhaps we just want to loosen the recommendation in the implementation note?

From:

> `roughnessFactor` is `1`.

to something like:

> `roughnessFactor` is `> 0.5`